apps/entry/serializers.py

```diff
@@ -375,6 +375,13 @@ def validate_figures(self, figures):
                                                   'not exist.')
         return figures
 
+    def validate_calculation_logic(self, value):
+        errors = OrderedDict()
+        if not value:
+            errors['calculation_logic'] = gettext('This field is required.')
+            raise serializers.ValidationError(errors)
+        return value
+
```

Comment:
We don't need errors here. This will create a nested error. {'calculation_logic': {'calculation_logic': 'errormessage'}}
Basically for each `validate_<field>`, DRF will create `key - value` pair.
NOTE: For errors in `validate`, this works (though raise is not required there).

```suggestion
    def validate_calculation_logic(self, value):
        if not value:
            raise serializers.ValidationError(_('This field is required'))
        return value

```

Reply:
@thenav56  Improved this one